cartridges/int_alma/cartridge/static/default/js/almaCheckout.js

```diff
@@ -265,4 +267,26 @@ window.addEventListener('DOMContentLoaded',
             errorMessagePosition.after(errorDiv);
         }
 
+        function displayFragmentOnFailure() {
+            var errorMessagePosition = document.querySelectorAll(almaContext.selector.fragmentErrors)[0];
+            var errorMessage = document.createTextNode("Alma is unable to process your payment. Please try again or reach out to us for assistance.");
+            var errorDiv = document.createElement('div');
+            errorDiv.classList.add('col-12', 'alma-error-message');
+            errorDiv.id = 'fragment-on-failure';
+            errorDiv.appendChild(errorMessage);
+            errorDiv.appendChild(document.createElement('br'));
+            errorMessagePosition.after(errorDiv);
+        }
+
+        function displayFragmentOnClose() {
+            var errorMessagePosition = document.querySelectorAll(almaContext.selector.fragmentErrors)[0];
+            var errorMessage = document.createTextNode("By closing the Alma popup, you have canceled the payment process.");
+            var errorDiv = document.createElement('div');
+            errorDiv.classList.add('col-12', 'alma-error-message');
+            errorDiv.id = 'fragment-on-close';
+            errorDiv.appendChild(errorMessage);
+            errorDiv.appendChild(document.createElement('br'));
+            errorMessagePosition.after(errorDiv);
```

Comment:
I think you can create a function to refactor.

In the next version, Jquery becomes essential for trigerring. wouldn't we use it for select? 
@syjust-alma WDYT?

Reply:
Refacto / DRY => ok for me 👌 

jQuery dependency =>

I think the jQuery part of the next version is an overridable part of the code (depending on how the integration will be done in the merchant environment => jQuery is mandatory only if the merchant theme is based on SFRA because the triggered event to listen is a jQuery event)

So, the rest of the cartridge should not depends on jQuery lib IMHO.